hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -674,15 +679,66 @@ public boolean isNearCacheEnabled() {
         return nearCacheConfig != null;
     }
 
+    /**
+     *
+     * @return
+     * @deprecated user {@link #getQueryCacheConfigs()} instead.
+     */
     public boolean isOptimizeQueries() {
-        return optimizeQueries;
+        return cacheDeserializedValues == CacheDeserializedValues.ALWAYS;
     }
 
+    /**
+     *
+     * Enable de-serialized value caching when evaluating predicates.
+     *
+     * @param optimizeQueries
+     * @return this {@code MapConfig} instance.
+     * @deprecated use {@link #setCacheDeserializedValues(CacheDeserializedValues)} instead
+     */
     public MapConfig setOptimizeQueries(boolean optimizeQueries) {
-        this.optimizeQueries = optimizeQueries;
+        if (optimizeQueries) {
```

Comment:
I am not sure whether (and how aggressively) we should try to detect conflict between (now deprecated) `optimizeQueries` and `CacheDeserializedValues` options